XenAdmin/Controls/Ballooning/VMMemoryControlsBasic.cs

```diff
@@ -100,21 +100,15 @@ protected override void OnPaint(PaintEventArgs e)
                     }
                     if (identical)
                     {
-                        switch (vm0.GetVirtualisationStatus)
-                        {
-                            case VM.VirtualisationStatus.OPTIMIZED:
-                                labelDMCUnavailable.Text = Messages.DMC_UNAVAILABLE_NOTSUPPORTED_PLURAL;
-                                break;
-                            case VM.VirtualisationStatus.PV_DRIVERS_NOT_INSTALLED:
-                                labelDMCUnavailable.Text = Messages.DMC_UNAVAILABLE_NOTOOLS_PLURAL;
-                                break;
-                            case VM.VirtualisationStatus.PV_DRIVERS_OUT_OF_DATE:
-                                labelDMCUnavailable.Text = Messages.DMC_UNAVAILABLE_OLDTOOLS_PLURAL;
-                                break;
-                            default:  // VM.VirtualisationStatus.UNKNOWN
-                                labelDMCUnavailable.Text = Messages.DMC_UNAVAILABLE_VMS;
-                                break;
-                        }
+                        var status = vm0.GetVirtualisationStatus;
+                        if (status.HasFlag(VM.VirtualisationStatus.IO_DRIVERS_INSTALLED | VM.VirtualisationStatus.MANAGEMENT_INSTALLED))

```

Comment:
Do you really need both I/O optimized and Management Installed for this, or is one or the other sufficient? My guess is that you only need one. But if you really need both, it might be better to make it more explicit with
HasFlag(IO_DRIVERS_INSTALLED) && HasFlag(MANAGEMENT_INSTALLED)